File: Assets/ApertureCutsceneEditor/Components/Actions/Action_SetMaterialFloat.js
#pragma strict
@script AddComponentMenu("Cutscene/Actions/Set Material Float")

@script ExecuteInEditMode()

@CutsceneEventOverrideNameAttribute("Set Material Float")

class Action_SetMaterialFloat extends CutsceneAction {
	var useMaterialOnObject:boolean = true;
	var materialIndex:int = 0;
	var material:Material;
	var propertyName:String = "_MyFloat";
	
	//#if !UNITY_FLASH
	@CutsceneActionAttribute("Set Float")
	//#endif
	function SetFloat(value:float) {
		SetupMaterial();
		if (!material) return;
		if (!material.HasProperty(propertyName)) return;
		material.SetFloat(propertyName,value);
	}
	
	@CutsceneEventExclude()
	function SetupMaterial() {
			if (useMaterialOnObject) {
			if (!Application.isPlaying) {
				if (renderer && renderer.sharedMaterials.length > materialIndex) {
					material = renderer.sharedMaterials[materialIndex];
				}
			}
			else {
				if (renderer && renderer.materials.length > materialIndex) {
					material = renderer.materials[materialIndex];
				}
			}
		}
	}
}

File: CutsceneEditor/CutsceneEditor/CutsceneEditor/Components/Actions/Action_SetMaterialFloat.cs
using UnityEngine;

[AddComponentMenu("Cutscene/Actions/Set Material Float")]

[ExecuteInEditMode()]

[CutsceneEventOverrideNameAttribute("Set Material Float")]

public class Action_SetMaterialFloat : CutsceneAction
{
    public bool useMaterialOnObject = true;
    public int materialIndex = 0;
    public Material material;
    public string propertyName = "_MyFloat";

    //#if !UNITY_FLASH
    [CutsceneActionAttribute("Set Float")]
    //#endif
    public void SetFloat(float value)
    {
        if (!material) return;
        if (!material.HasProperty(propertyName)) return;
        material.SetFloat(propertyName, value);
    }

    [CutsceneEventExclude()]
    public void SetupMaterial()
    {
        if (useMaterialOnObject)
        {
            if (!Application.isPlaying)
            {
                if (renderer && renderer.sharedMaterials.Length > materialIndex)
                {
                    material = renderer.sharedMaterials[materialIndex];
                }
            }
            else
            {
                if (renderer && renderer.materials.Length > materialIndex)
                {
                    material = renderer.materials[materialIndex];
                }
            }
        }
    }
}

